__tests__/app.test.ts

```diff
@@ -277,5 +277,70 @@ describe('🧪 Express Application', () => {
 				return request(app).post('/api/cats/update').send(data).expect(401);
 			});
 		});
+
+		describe('GET /api/cats/leaderboard/:range', () => {
+			it('400: should return when passed an invalid date range', () => {
+				return request(app)
+					.get('/api/cats/leaderboard/INVALID')
+					.expect(400)
+					.then(({ body: { success, message } }) => {
+						expect(success).toBe(false);
+						expect(message).toBe("'invalid' is not a recognized range");
+					});
+			});
+			describe('RANGE', () => {
+				it('200: DAILY', () => {
+					return request(app)
+						.get('/api/cats/leaderboard/DAILY')
+						.expect(200)
+						.then(({ body: { success, data, range } }) => {
+							expect(success).toBe(true);
+							expect(range).toBe('daily');
+							expect(data).toBeSorted();
+							expect(data.length).toBe(2);
+						});
```

Comment:
Did you migrate and seed? It passes the test on the GitHub action and locally for me

Reply:
Yep, same issue. (And shouldn't your pre-test be sorting that anyway?)

Reply:
Due to an outstanding issue in Prisma, there's no true programmatic way to reset a database for tests without a slightly hacky method involving running a child process and executing a command via JS. This was actually implemented in [this PR](https://github.com/clowder-cloud/backend/pull/21).

As per why it isn't working for you, I really couldn't say. I destroyed my container, deleted the volumes, re-cloned and it still passes the test for me, and also passes on the GitHub action.

Reply:
After tearing down docker to the point I even removed the PostgreSQL image, I re-installed everything, ran seed, ran tests, ran tests (to make sure it wasn't an issue with re-seeding), looked at Prisma Studio and I can still say I haven't ran into a single issue. I know you've said you have, but are you 100% sure you've reseeded?

![image](https://github.com/user-attachments/assets/75760795-870b-400a-bcda-f0f6e17b8a65)

Reply:
```
npm i
npm run migrate
npm run seed
npm t
```

Is that right?

I could try rebuilding the Docker, but I feel that would just be hiding the underlying error somewhere.

Reply:
Do you have a clean working directory when you run the tests?

Reply:
I've done every single thing I can think of - I've tested it in an environment I was working in, a fresh install, GitHub actions does everything from fresh. I really can't see how this would be an issue with the test.

Reply:
OK, I think I know the problem.... You're testing a fixed data set, but the test involves looking at timestamps from the last 24 hours.... that's dependent on when you run the test.

Reply:
(I currently get a result of 1, where as before it was 2)

Reply:
Okay, I'll re-write the seed file tomorrow morning to dynamically set the timestamps.

Reply:
You could fix it by checking the runtime environment, and if it's testing then the function could use a fixed timestamp and pretend that's today.

Reply:
```
new Date(
  (process.env.NODE_ENV === "test")
    ? A_FIXED_TIMESTAMP
    : null
)

Reply:
If you want to do that tonight, you're more than welcome ;) I'm hoping in my lovely warm bed now

Reply:
I'm worried we'll run out of reviewers, so I don't want to commit on the PR.